source/apiVolontaria/apiVolontaria/tests/tests_view_Users.py

```diff
@@ -0,0 +1,162 @@
+            'password': 'test123!',
+        }
+
+        response = self.client.post(
+            reverse('users'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        user = User.objects.get(username="John")
+        activation_token = ActivationToken.objects.filter(user=user)
+
+        self.assertEqual(1, len(activation_token))
+
+    def test_create_new_user_without_username(self):
+        """
+        Ensure we can't create a new user without username
+        """
+        data = {
+            'email': 'John@mailinator.com',
+            'password': 'test123!',
+        }
+
+        response = self.client.post(
+            reverse('users'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {"username": ["This field is required."]}
+        self.assertEqual(json.loads(response.content), content)
+
+    def test_create_new_user_without_email(self):
+        """
+        Ensure we can't create a new user without email
+        """
+        data = {
+            'username': 'John',
+            'password': 'test123!',
+        }
+
+        response = self.client.post(
+            reverse('users'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {"email": ["This field is required."]}
+        self.assertEqual(json.loads(response.content), content)
+
+    def test_create_new_user_without_password(self):
+        """
+        Ensure we can't create a new user without password
+        """
+        data = {
+            'username': 'John',
+            'email': 'John@mailinator.com',
+        }
+
+        response = self.client.post(
+            reverse('users'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {"password": ["This field is required."]}
+        self.assertEqual(json.loads(response.content), content)
+
+    def test_list_users(self):
+        """
+        Ensure we can list all users.
+        """
+        self.client.force_authenticate(user=self.admin)
+
+        response = self.client.get(reverse('users'))
+
+        self.assertEqual(json.loads(response.content)['count'], 2)
+
+        first_user = json.loads(response.content)['results'][0]
+        self.assertEqual(first_user['id'], self.user.id)
+
+        # Check the system doesn't return attributes not expected
+        attributes = ['id', 'username', 'email', 'first_name',
+                      'last_name', 'is_active']
+        for key in first_user.keys():
+            self.assertTrue(
+                key in attributes,
+                'Attribute "{0}" is not expected but is '
+                'return by the system.'.format(key)
```

Comment:
'Attribute "{0}" is not expected but is '
'**returned** by the system.'.format(key)